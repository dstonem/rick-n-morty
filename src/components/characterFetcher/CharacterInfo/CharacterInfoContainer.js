export default function CharacterInfoContainer({ source, name, status }) {
  return (
    <div>
      <img src={source} alt={name} data-testid="img" />
      <h2>{name}</h2>
      <p>Status: {status}</p>
    </div>
  );
}
