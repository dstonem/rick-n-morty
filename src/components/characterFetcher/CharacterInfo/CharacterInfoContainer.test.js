import React from "react";
import { cleanup, render } from "@testing-library/react";
import CharacterInfoContainer from "./CharacterInfoContainer";

afterEach(cleanup);

const props = {
  name: "rick",
  source: "src",
  status: "alive",
};

describe("CharacterInfoContainer", () => {
  it("renders correctly", () => {
    const { asFragment, getByText, getByTestId } = render(
      <CharacterInfoContainer {...props} />
    );
    expect(getByText("rick")).toBeTruthy();
    expect(getByTestId("img")).toHaveAttribute("src", "src");
    expect(getByText("Status: alive")).toBeTruthy();

    expect(asFragment()).toMatchSnapshot();
  });
});
