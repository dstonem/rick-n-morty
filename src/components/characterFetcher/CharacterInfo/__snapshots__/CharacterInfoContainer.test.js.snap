// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterInfoContainer renders correctly 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="rick"
      data-testid="img"
      src="src"
    />
    <h2>
      rick
    </h2>
    <p>
      Status: alive
    </p>
  </div>
</DocumentFragment>
`;
